rust-runtime/aws-smithy-http-server/src/protocols.rs

```diff
@@ -66,17 +66,18 @@ pub fn content_type_header_classifier(
     Ok(())
 }
 
-pub fn accept_header_classifier(headers: &HeaderMap, content_type: &'static str) -> bool {
+pub fn accept_header_classifier(headers: &HeaderMap, content_type: &'static str) -> Result<(), NotAcceptableReason> {
     if !headers.contains_key(http::header::ACCEPT) {
-        return true;
+        return Ok(());
     }
-    // Must be of the form: type/subtype
+    // Must be of the form: `type/subtype`.
     let content_type = content_type
```

Comment:
I looked at `Mime::parse` and it appears to be a [non-negligible process](https://github.com/hyperium/mime/blob/ce5062d216bf757a0ed3fc70f0fe255d1c8d74ae/src/parse.rs#L111-L182), so putting `Mime` behind a `OnceCell` and incurring an atomic read should indeed speed things up.

I also learnt that [`mime` is not actively maintained](https://github.com/hyperium/mime/issues/142), and stumbled upon [`mediatype`](https://docs.rs/mediatype/0.19.13/mediatype/), which is const-constructible, so for our static mime type coming from the sSDK I think it would make more sense to switch to that. It's also zero-copy, so parsing the request's `Accept` and `Content-Type` headers' mime types and performing [the type + subtype equality check](https://docs.rs/mediatype/0.19.13/src/mediatype/name.rs.html#59-63) should be faster.

Reply:
Sounds like a plan

Reply:
For the first part: https://github.com/awslabs/smithy-rs/pull/2629

Reply:
#2629 is throwaway work if we switch away from `mime`. I've opened https://github.com/awslabs/smithy-rs/issues/2666 to track that.